File: UniFarm/Capstone.UniFarm.Domain/Models/Transfer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Capstone.UniFarm.Domain.Models
{
    [Table("Transfer")]
    public partial class Transfer
    {
        public Transfer()
        {
            Orders = new HashSet<Order>();
        }
        [Key]
        public Guid Id { get; set; }
        public Guid CollectedId { get; set; }
        public Guid StationId { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public DateTime? ReceivedDate { get; set; }
        public Guid? CreatedBy { get; set; }
        public Guid? UpdatedBy { get; set; }
        public string? NoteSend { get; set; }
        public string? NoteReceived { get; set; }
        public string? Code { get; set; }
        [StringLength(100)]
        public string? Status { get; set; }
        
        [ForeignKey(nameof(CollectedId))]
        [InverseProperty(nameof(CollectedHub.Transfers))]
        public virtual CollectedHub Collected { get; set; } = null!;

        [ForeignKey(nameof(StationId))]
        [InverseProperty("Transfers")]
        public virtual Station Station { get; set; } = null!;
        public virtual ICollection<Order>? Orders { get; set; }
    }
}

File: FTAFE/src/core/models/transfer.ts
import { CollectedHub, Station } from './staff';

export type Transfer = {
    id: string;
    collectedId: string;
    stationId: string;
    createdAt: string;
    updatedAt: string;
    expectedReceiveDate: string | null;
    receivedDate: string;
    createdBy: string;
    updatedBy: string | null;
    noteSend: string;
    noteReceived: string;
    code: string;
    status: string;
    collected: CollectedHub;
    station: Station;
    orders: any | null;
};
export interface CreateTransferForm {
    collectedId: string;
    stationId: string;
    noteSend: string;
    expectedReceiveDate: Date | string;
    orderIds: string[];
}

export interface UpdateTransferForm {
    id: string;
    status: string;
    noteReceived: string;
}
